docs/source/en/chat_templating.md

```diff
@@ -0,0 +1,187 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
```

Comment:
Generally, I'm wondering a bit whether this page would better fit as a new sub-task page under the "Generation" task section as "chat" here:
https://huggingface.co/docs/transformers/generation_strategies

Reply:
Don't fully agree here I guess since we already have training sections in sub-tasks see: https://huggingface.co/docs/transformers/tasks/language_modeling#train

Think our task guides are not just limited to inference.

At the moment, I don't see how "chat" would be used for a task outside of text so it's quite natural for me to expect "chat" below "text-generation" as a sub-task.

But not a big deal I think so ok for me to leave as is for now! 

Cc'ing @gante here though just in case

Reply:
Chat would definitely be used outside pure text, e.g. for multi-round conversations with multimodal models like IDEFICS/Flamingo!

Reply:
Eventually once the refactor of the generation docs is a bit more underway I could see this doc living there, but for now having it live in the developer guides sounds good to me.